controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -234,6 +245,24 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// ClusterRegistryConfig represents registry config used with the cluster.
 	// +optional
 	ClusterRegistryConfig *RegistryConfig `json:"clusterRegistryConfig,omitempty"`
+
+	// AutoNode set the AutoNode mode and AutoNode role ARN.
+	// +optional
+	AutoNode *AutoNode `json:"autoNode,omitempty"`
+}
+
+// AutoNode set the AutoNode mode and AutoNode role ARN.
+type AutoNode struct {
+	// AutoNode mode allowed values are enabled & disabled
+	// +kubebuilder:validation:Enum=enabled;disabled
+	// +kubebuilder:default=disabled
+	// +optional
```

Comment:
```suggestion
	// AutoNode mode specifies the mode for the AutoNode setting.
	// +kubebuilder:validation:Enum=Enabled;Disabled
	// +kubebuilder:default=Disabled
	// +optional
```

Reply:
Looks like this has been updated, marking as resolved.